frontend/packages/metal3-plugin/src/k8s/requests/bare-metal-host.ts

```diff
@@ -13,3 +17,22 @@ export const createBareMetalHost = async (bareMetalHost, secret) => [
   await k8sCreate(SecretModel, secret),
   await k8sCreate(BareMetalHostModel, bareMetalHost),
 ];
+
+export const deprovision = async (machine: MachineKind, machineSet: MachineSetKind) => {
```

Comment:
machineset is optional here:
If host's machine is managed by machineSet -> annotate the machine and scale down the machineset to trigger deprovisioning.
if host's machine is not managed by machineSet (e.g. master) -> just delete the machine to start deprovisioning.

Reply:
fixed

I am still adding the annotation to distinguish the machines for hiding the deprovision action